lib/puppet/defaults.rb

```diff
@@ -961,6 +961,18 @@ def self.default_vendormoduledir
           certificate revocation checking and does not attempt to download the CRL.
         EOT
     },
+    :key_type => {
+      :default => 'rsa',
+      :type    => :enum,
+      :values  => %w[rsa ec],
+      :desc    => "The type of private key. Valid values are `rsa` and `ec`. Default is `rsa`."
+    },
+    :named_curve => {
+      :default => 'prime256v1',
+      :type    => :string,
+      :desc    => "The short name for the EC curve. Valid values must be one of the curves in
+                  `OpenSSL::PKey::EC.builtin_curves`. Default is `prime256v1`."
+    },
```

Comment:
Should specify this is the curve for the generated private and public key pair, as opposed to the curve used in the TLS connection (for example when using `TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA` ciphersuite).

Reply:
Done